examples/00_datatypes_and_datasets.ipynb

```diff
@@ -0,0 +1,1084 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_raise(y.reshape(-1, 1), mtype=\"np.ndarray\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For use in own code or additional metadata, the error message can be obtained using the `check_is` function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sktime.datatypes import check_is\n",
+    "\n",
+    "check_is(y, mtype=\"np.ndarray\", return_metadata=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "and metadata is produced if the argument passes the validity check:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_is(y.reshape(-1, 1), mtype=\"np.ndarray\", return_metadata=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: if the name of the mtype is ambiguous and can refer to multiple scitypes, the additional argument `scitype` must be provided. This should not be the case for any common in-memory containers, we mention this for completeness."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "check_is(y, mtype=\"np.ndarray\", scitype=\"Series\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Section 2.1.2: validity checking, example 2 (non-obvious mistake)\r\n",
+    "\r\n",
+    "Suppose we have converted our data into a multi-index panel, i.e., we want to have a `Panel` of mtype `pd-multiindex`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "cols = [\"instances\", \"time points\"] + [f\"var_{i}\" for i in range(2)]\n",
+    "X = pd.concat(\n",
+    "    [\n",
+    "        pd.DataFrame([[0, 0, 1, 4], [0, 1, 2, 5], [0, 2, 3, 6]], columns=cols),\n",
+    "        pd.DataFrame([[1, 0, 1, 4], [1, 1, 2, 55], [1, 2, 3, 6]], columns=cols),\n",
+    "        pd.DataFrame([[2, 0, 1, 42], [2, 1, 2, 5], [2, 2, 3, 6]], columns=cols),\n",
+    "    ]\n",
+    ").set_index([\"instances\", \"time points\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is not obvious whether `X` satisfies the `pd-multiindex` specification, so let's check:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sktime.datatypes import check_raise\n",
+    "\n",
+    "# uncomment this to see the informative error message\n",
+    "#\n",
+    "# check_raise(X, mtype=\"pd-multiindex\")"
```

Comment:
no, because then it will raise an informative error message, which will then be caught by CI `test_examples`, which thinks that the notebooks breaks (as opposed to breaking in a place where it's supposed to).

unless you have a better idea how to handle this, of course

Reply:
Oh, I see. At my work, I saw somebody getting around this with a try except block, and in the except block you can print the error message somehow. I will try to find it

Reply:
hmmmm - but wouldn't this confuse notebook users since they think the try/except is part of it?

Reply:
Yes, try/except could confuse users... looks like there is no ideal option. The issue with current setup is that things that require uncommenting tend to be optional in tutorials I have seen.

Maybe have the instruction to uncomment the cell be a part of the markdown, rather than a comment itself?

Reply:
yes, agreed, text in comments is more likely to be ignored, so I moved the instruction to the markdown where it will be percieved as part of the "voiceover"